sympy/ntheory/generate.py

```diff
@@ -246,9 +263,32 @@ def primepi(n):
     n = int(n)
     if n < 2:
         return 0
-    else:
-        return sieve.search(n)[0]
-
+    lim = int(n ** 0.5)
+    lim -= 1
+    lim = max(lim,0)
+    while lim * lim <= n:
+        lim+=1
+    lim-=1
+    arr1 = [0]*(lim+1)
+    arr2 = [0]*(lim+1)
+    for i in range(1, lim+1):
+        arr1[i] = i-1
+        arr2[i] = 0 if i==0 else n//i - 1
```

Comment:
Not its not. But not testing won't improve time taken, because the complexity of algorithm is about `O(n**0.75)` , whereas `i==0` is tested only `sqrt(n)` times